packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -730,9 +730,12 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
                 const contextSummary = await context.summarize(true /* fullTree */, trackState);
                 summaryBuilder.addWithStats(contextId, contextSummary);
 
-                // Prefix the child's id to the ids of its GC nodes. This gradually builds the id of each node
-                // to be a path from the root.
-                gcDataBuilder.prefixAndAddNodes(contextId, contextSummary.gcData.gcNodes);
+                // back-compat: 0.40 - We don't use the gcData returned by summarize anymore.
```

Comment:
I do not think there is any back compat considerations here, as data store and DDSs are within same layer.
When do we get undefined here?

Reply:
Same question for code below

Reply:
We won't get undefined here but I made gcData optional in the interface so the compiler complains. The reason I made it optional is because I wanted to remove it from the container-runtime layer but keep it here.

Reply:
Given that the concerns do not cross from one (compat) layer to another, I'm not sure why what happens in next release matters. I.e. if its non-conditional in this release, then why mark it as conditional? This does not change the fact that you can remove it next release.

Reply:
I wanted to make it semi-conditional in this release. Basically, `DataStores`, `ContainerRuntime` and `DataStoreContext` doesn't return gcData in its summarize methods. `DataStoreRuntime` and `RemoteChannelContext` returns it.
I could just keep the code that returns it everywhere and just not use it. But does it really matter? Does making it conditional have any side effects?